### DISCUSSION TITLE
is HTML exported sanitized?

### DISCUSSION THREAD

**Discussion by toddteller** [facebook/lexical#4534] [Q&A] [answered]
The html got from `const htmlString = $generateHtmlFromNodes(editor, selection | null);` is sanitized?

**Comment by reactuz** [comment]
@acywatson @fantactuka @thegreatercurve @tylerjbainbridge @ivailop7 @zurfyx 
I also need an answer to this. Please, say your word on XSS.

**Reply by ivailop7** [reply, marked as answer]
While there are certainly some good practices like [ignoring the 'script' tag when parsing the HTML:](https://github.com/facebook/lexical/blob/a82b7547ea158a5472afda0f702ed844ca25fd0a/packages/lexical-html/src/index.ts#L163)  
and [some limited URL sanitization](https://github.com/facebook/lexical/blob/a82b7547ea158a5472afda0f702ed844ca25fd0a/packages/lexical-playground/src/utils/url.ts#L17).

For all intended purposes, in my understanding, the HTML is not sanitized. The core nodes do use safe tags, but given the flexibility the decorator node gives, it should be the user's responsibility for due diligence of the plugins you are adding in your editor.

**Reply by reactuz** [reply]
@toddteller Does this also answer your question? I'm satisfied with this one.

**Reply by ys-oo** [reply]
how can i use dompurify to make sure the html is sanitized in a way that make it safe to store and safe for collaborative editing ? 

thank you <3
